### PR TITLE
MTY_Archive module

### PR DESCRIPTION
**Pull request by Namaneo**
This PR adds a module in charge of manipulating archive files. It currently only supports extraction of ZIP archive. Since `miniz` is used under the hood, this library has been updated to the upstream version, with additional modifications listed between the lines `115 -> 126` of `deps/miniz.h`.

**Reason**: with the `MTY_Webview` module coming, it will be convenient to download or read a single bundle file containing the whole web application, and extract it in memory.

### CODE REVIEW

File: src/archive.c [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	MTY_Free(ctx);
+	*archive = NULL;
+}
+
+bool MTY_ArchiveReadFile(MTY_Archive *ctx, MTY_ArchiveFile **file)
+{
+	if (*file == NULL) {
+		ctx->current_file = MTY_ListGetFirst(ctx->files);
+
+	} else if (ctx->current_file) {
```
Comment (Kodikuu) [inline src/archive.c]:
```suggestion
bool MTY_ArchiveReadFile(MTY_Archive *ctx, MTY_ArchiveFile **file)
{
	if (!ctx || !file)
		return false;
	
	if (*file == NULL) {
		ctx->current_file = MTY_ListGetFirst(ctx->files);

	} else if (ctx->current_file) {
```

File: src/archive.c [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+	*file = ctx->current_file ? ctx->current_file->value : NULL;
+
+	return *file != NULL;
+}
+
+bool MTY_ArchiveUnpack(MTY_Archive *ctx, MTY_ArchiveFile *file)
+{
+	file->data = mz_zip_reader_extract_file_to_heap(ctx->zip, file->path, NULL, 0);
+
```
Comment (Kodikuu) [inline src/archive.c]:
```suggestion
bool MTY_ArchiveUnpack(MTY_Archive *ctx, MTY_ArchiveFile *file)
{
	if (!ctx || !file)
		return false;

	file->data = mz_zip_reader_extract_file_to_heap(ctx->zip, file->path, NULL, 0);

```

File: src/archive.c [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	mz_zip_archive *zip;
+	MTY_List *files;
+	MTY_ListNode *current_file;
+};
+
+MTY_Archive *MTY_ArchiveOpen(const void *data, size_t size)
+{
+	MTY_Archive *ctx = MTY_Alloc(1, sizeof(MTY_Archive));
+
+	ctx->zip = MTY_Alloc(1, sizeof(mz_zip_archive));
```
Comment (Kodikuu) [inline src/archive.c]:
```suggestion
MTY_Archive *MTY_ArchiveOpen(const void *data, size_t size)
{
	if (!data)
		return NULL;

	MTY_Archive *ctx = MTY_Alloc(1, sizeof(MTY_Archive));

	ctx->zip = MTY_Alloc(1, sizeof(mz_zip_archive));
```

**Review by Kodikuu — CHANGES_REQUESTED** [review]

**Review by Kodikuu — APPROVED** [review]

**Comment by Namaneo** [comment]
Theorically useless for now, closing.

**Pull request closed without merging**